### ISSUE TITLE
Exception Handling

### ISSUE THREAD

**Issue by zainozzaini** [spatie/laravel-uptime-monitor#209]
I got this error via console on php artisan monitor:check-uptime

```
Argument 1 passed to Spatie\UptimeMonitor\MonitorCollection::Spatie\UptimeMonitor\{closure}() must be an instance of Spatie\UptimeMonitor\Exception, instance of GuzzleHttp\Exception\ConnectException given

  at vendor/spatie/laravel-uptime-monitor/src/MonitorCollection.php:30
    26| 
    27|                 $monitor->uptimeRequestSucceeded($response);
    28|             },
    29| 
   30|             'rejected' => function (RequestException $exception, $index) {
    31|                 $monitor = $this->getMonitorAtIndex($index);
    32| 
    33|                 ConsoleOutput::error("Could not reach {$monitor->url} error: `{$exception->getMessage()}`");
    34| 
```

Here is the monitor list if i not set exception type on line 30.

![image](https://user-images.githubusercontent.com/3208540/97292183-e49acf80-1885-11eb-92ae-5008a5dd7614.png)

**Comment by noahfrederick** [comment]
This is a duplicate of https://github.com/spatie/laravel-uptime-monitor/issues/210 and is potentially fixed by https://github.com/spatie/laravel-uptime-monitor/pull/208.

**Comment by nilsringersma** [comment]
I can confirm this issue. 

**Problem**
When the Promise for the monitor is rejected with a [GuzzleHttp\Exception\ConnectException](https://github.com/guzzle/guzzle/blob/master/src/Exception/ConnectException.php) execution fails because Argument 1 passed to Spatie\UptimeMonitor\MonitorCollection::Spatie\UptimeMonitor\{closure}() must be an instance of GuzzleHttp\Exception\RequestException.

**Reproduce**
Add a monitor to a non-existing URL and run the artisan command monitor:check-uptime.

**Comment by Nielsvanpach** [comment]
I think this is related to https://github.com/spatie/laravel-uptime-monitor/pull/208 and fixed in v4.0.1
Feel free to ping me again if the issues has not been solved yet.
